src/mslice/cli/helperfunctions.py

```diff
@@ -75,6 +76,12 @@ def _process_axis(axis, fallback_index, input_workspace, string_function=_string
     # check to see if axis is just a name e.g 'DeltaE' or a full binning spec e.g. 'DeltaE,0,1,100'
     if ',' in axis:
         axis = string_function(axis)
+        if fallback_index != 0:
+            return axis
+        x_step = get_axis_range(input_workspace, axis.units)[-1]
```

Comment:
Instead of using the last element of the tuple you could try using the step element directly ([2]) and catch the error if this is not possible. Just in case get_axis_range does not return the expected tuple...